### PR TITLE
Bluetooth: CAP: Shell: Add cancel command

### PR DESCRIPTION
**Pull request by kruithofa**
We didn't have a cancel command for the CAP commander, so adding this command here.

### CODE REVIEW

File: subsys/bluetooth/audio/shell/cap_commander.c [hunk 1]
```diff
@@ -103,6 +103,19 @@ static struct bt_cap_commander_cb cbs = {
 #endif /* CONFIG_BT_MICP_MIC_CTLR */
 };
 
+static int cmd_cap_commander_cancel(const struct shell *sh, size_t argc, char *argv[])
+{
+	int err = 0;
+
+	err = bt_cap_commander_cancel();
+	if (err != 0) {
+		shell_print(sh, "Failed to cancel unicast audio procedure: %d", err);
```
Comment (Thalley) [inline subsys/bluetooth/audio/shell/cap_commander.c:112]:
```suggestion
		shell_print(sh, "Failed to cancel CAP commander procedure: %d", err);
```

File: subsys/bluetooth/audio/shell/cap_commander.c [hunk 2]
```diff
@@ -103,6 +103,19 @@ static struct bt_cap_commander_cb cbs = {
 #endif /* CONFIG_BT_MICP_MIC_CTLR */
 };
 
+static int cmd_cap_commander_cancel(const struct shell *sh, size_t argc, char *argv[])
+{
+	int err = 0;
```
Comment (Thalley) [inline subsys/bluetooth/audio/shell/cap_commander.c:108]:
```suggestion
	int err;
```

**Review by Thalley — CHANGES_REQUESTED** [review]
Minor request, otherwise LGTM. 

We should add a BSIM test for this as well (in a separate change)

**Review by Thalley — CHANGES_REQUESTED** [review]
Please also add it to https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/connectivity/bluetooth/api/shell/cap.rst

**Review by Thalley — APPROVED** [review]

**Review by larsgk — APPROVED** [review]

**Pull request merged**